code/__DEFINES/_helpers.dm

```diff
@@ -13,3 +13,10 @@
 
 /// subtypesof(), typesof() without the parent path
 #define subtypesof(typepath) ( typesof(typepath) - typepath )
+
+/// Takes a datum as input, returns its ref string, or a cached version of it
+/// This allows us to cache \ref creation, which ensures it'll only ever happen once per datum, saving string tree time
+/// It is slightly less optimal then a []'d datum, but the cost is massively outweighed by the potential savings
+/// It will only work for datums mind, for datum reasons
+/// : because of the embedded typecheck
+#define text_ref(datum) (isdatum(datum) ? (datum:cached_ref ||= "\ref[datum]") : ("\ref[datum]"))
```

Comment:
Do me a favor. If you change `cached_ref` to a different key (one that is valid but not on every datum, so it compiles).

Do unit tests catch it? I want to make sure that even if we happen to have a `cached_ref` somewhere that the `:` breaking would still be caught

Reply:
it would runtime yeah, since this code is hit for everything we qdel, and we qdel almost everything in testing

Reply:
ok i thought so